analyzer/pubclient/pubclient.go

```diff
@@ -0,0 +1,92 @@
+package pubclient
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"syscall"
+	"time"
+
+	coreCommon "github.com/oasisprotocol/oasis-core/go/common"
+)
+
+var permittedNetworks = map[string]bool{
+	"tcp4": true,
+	"tcp6": true,
+}
+
+type NotPermittedError struct {
+	// Note: .error is the implementation of .Error, .Unwrap etc. It is not
+	// in the Unwrap chain. Use something like
+	// `EVMDeterministicError{fmt.Errorf("...: %w", err)}` to set up an
```

Comment:
```suggestion
	// `NotPermittedError{fmt.Errorf("...: %w", err)}` to set up an
```

btw, what's the reasoning for this?

Reply:
oops yeah

Reply:
fixed